### PR TITLE
fix: auth map initial

### LINKED ISSUE #1: AuthMap: allow a map of auth objects and reference in each RestAction
this will allow any number of strategies to be executed against any baseUrl + endpoint

### PR DESCRIPTION
**Pull request by dnitsch**
Addresses #1

### CODE REVIEW

File: pkg/rest/auth.go [hunk 1]
```diff
@@ -24,39 +24,71 @@ type ConfigOAuth struct {
...
-	config           AuthConfig
-	oAuthConfig      *clientcredentials.Config
-	basicAuthToToken string
+type AuthMap map[string]AuthConfig
+
+// type Auth map[string]struct {
+// 	config           AuthMap
+// 	oAuthConfig      *clientcredentials.Config
+// 	basicAuthToToken string
+// }
```
Comment (ElvenSpellmaker) [inline pkg/rest/auth.go]: Is this meant to be commented out here?

File: pkg/rest/rest.go [hunk 2]
```diff
@@ -77,6 +63,29 @@ type Action struct {
...
+func (a *Action) WithHeader() *Action {
+	suppliedHeader := a.HttpHeaders
+	h := &http.Header{}
+	// set default values
+	h.Add("Accept", "application/json")
+	h.Add("Content-Type", "application/json")
+	// overwrite or add additional header attributes on a per action basis
+	if suppliedHeader != nil {
+		for k, v := range *suppliedHeader {
+			h.Add(k, v)
```
Comment (ElvenSpellmaker) [inline pkg/rest/rest.go:83]: Out of interest do you standardise the casing anywhere? If not what happens if someone supplies `accept` vs `Accept`, would both headers be present?

File: strategyrestseeder.go [hunk 3]
```diff
@@ -74,16 +73,15 @@ func (s *StrategyRestSeeder) WithRestClient(rc rest.Client) *StrategyRestSeeder
...
-		}
-	}
-	s.rest = s.rest.WithHeader(httpHeader)
+	// httpHeader := &http.Header{}
+	// if len(ra.HttpHeaders) > 0 {
+	// 	for k, v := range ra.HttpHeaders {
+	// 		httpHeader.Add(k, v)
+	// 	}
+	// }
+	// s.rest = s.rest.WithHeader(httpHeader)
```
Comment (ElvenSpellmaker) [inline strategyrestseeder.go]: Remove this as you've put it elsewhere?

File: pkg/rest/auth.go [hunk 4]
```diff
@@ -24,39 +24,71 @@ type ConfigOAuth struct {
...
+}
+
+type basicToToken struct {
+	username     string
+	password     string
+	headerKey    string
+	headerValFmt string
+}
+
+type authMap map[string]auth
```
Comment (paulholiday) [inline pkg/rest/auth.go]: Slightly confusing having two different types `AuthMap` and `authMap`

File: pkg/rest/rest.go [hunk 5]
```diff
@@ -77,6 +63,29 @@ type Action struct {
 	PayloadTemplate      string             `yaml:"payloadTemplate"`
 	Variables            map[string]any     `yaml:"variables"`
 	RuntimeVars          *map[string]string `yaml:"runtimeVars,omitempty"`
+	AuthMapRef           string             `yamls:"authMapRef"`
```
Comment (paulholiday) [inline pkg/rest/rest.go]: Is yamls a thing?

File: pkg/rest/rest.go [hunk 6]
```diff
@@ -94,18 +103,19 @@ const (
 // do performs all the network calls -
```
Comment (paulholiday) [inline pkg/rest/rest.go:103]: If you have to write this comment then probably then the function should be renamed

File: pkg/rest/rest.go (hunk 2 above)
Comment (dnitsch) [reply pkg/rest/rest.go:83]: good question - I think the end result would be something like `map[string][]string{"Accept": []string{"application/json", "application/xml"}}`  as that is the actual type of the `http.Header`

File: pkg/rest/auth.go (hunk 4 above)
Comment (dnitsch) [reply pkg/rest/auth.go]: name changed...

**Pull request merged**